### PR TITLE
refactor(lodash): orderBy

### PR DESCRIPTION
**Pull request by Haroenv**
The utility is copied from the helper (which passed all lodash tests), I put it inline in the connector, since it's not used in any other spot.

IFW-746

### CODE REVIEW

File: packages/react-instantsearch-core/src/connectors/connectMenu.js [hunk 1]
```diff
@@ -31,6 +30,82 @@ function getCurrentRefinement(props, searchState, context) {
...
+    }
+
+    // This ensures a stable sort in V8 and other engines.
+    // See https://bugs.chromium.org/p/v8/issues/detail?id=90 for more details.
+    return object.index - other.index;
+  });
+
+  return result.map(res => res.value);
+}
+
```
Comment (tkrugg) [inline packages/react-instantsearch-core/src/connectors/connectMenu.js]: could we copy the tests too?
it's complex enough that it might be worth it to "privately export" it and throughly test it.

Comment (Haroenv) [reply packages/react-instantsearch-core/src/connectors/connectMenu.js]: It’s an option, since I don’t think anyone will try to look for this in this particular file

Comment (samouss) [reply packages/react-instantsearch-core/src/connectors/connectMenu.js]: The call to the `orderBy` is not that useful I think. We use the function to sort only when the SFFV is allowed **AND** the facet values are **NOT** from the SFFV. Which means that we can use the `sortBy` option of `getFacetValues` if we really have to. WDYT?

Comment (tkrugg) [reply packages/react-instantsearch-core/src/connectors/connectMenu.js]: That sounds good as long as it's not a breaking change. The less code the better. 😄

Comment (Haroenv) [reply packages/react-instantsearch-core/src/connectors/connectMenu.js]: changed in adbe458

File: packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js [hunk 2]
```diff
@@ -424,16 +424,16 @@ describe('connectMenu', () => {
...
         {
           name: 'oy',
           isRefined: true,
           count: 10,
         },
+        {
+          name: 'wat',
+          isRefined: false,
+          count: 20,
+        },
```
Comment (tkrugg) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js]: I guess since now we no longer "order the facets by `isRefined`" but rather let getFacetValues handle it, the description of the test should change to something that "uses the getFacetValue default order.
And check that we're passing it `undefined` when `searchable && !isFromSearch`.

Comment (Haroenv) [reply packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js]: Makes sense, I’ll change the test description

File: packages/react-instantsearch-core/src/core/indexUtils.js [hunk 3]
```diff
@@ -8,13 +8,20 @@ export function getIndexId(context) {
...
+  if (searchResults.results.hits) {
+    return searchResults.results;
   }
+
+  const indexId = getIndexId(context);
+  if (searchResults.results[indexId]) {
+    return searchResults.results[indexId];
+  }
+
+  return null;
```
Comment (samouss) [inline packages/react-instantsearch-core/src/core/indexUtils.js]: We could avoid the double `null` return with the condition `searchResults.results` inline: 

```js
export function getResults(searchResults, context) {
  if (searchResults.results && searchResults.results.hits) {
    return searchResults.results;
  }

  const indexId = getIndexId(context);
  if (searchResults.results && searchResults.results[indexId]) {
    return searchResults.results[indexId];
  }

  return null;
}
```

IMO it's easier to read but I let you chose which one you want to implement.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js (hunk 2 above)
Comment (samouss) [reply packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js]: You forget to change the name of the test?

File: packages/react-instantsearch-core/src/core/indexUtils.js (hunk 3 above)
Comment (Haroenv) [reply packages/react-instantsearch-core/src/core/indexUtils.js]: yes, thought of that as an alternative, went for this style because it's less comparisons

Comment (Haroenv) [reply packages/react-instantsearch-core/src/core/indexUtils.js]:
```js
export function getResults(searchResults, context) {
  if (searchResults.results) {
    if (searchResults.results.hits) {
      return searchResults.results;
    }

    const indexId = getIndexId(context);
    if (searchResults.results[indexId]) {
      return searchResults.results[indexId];
    }
  }

  return null;
}
```

this is also somewhat easier since it has a return less, but all three styles suggested are fine I think

File: packages/react-instantsearch-core/src/connectors/connectMenu.js [hunk 4]
```diff
@@ -146,28 +145,23 @@ export default createConnector({
...
-
-    const sortedItems =
-      searchable && !isFromSearch
-        ? orderBy(
-            items,
-            ['isRefined', 'count', 'label'],
-            ['desc', 'desc', 'asc']
-          )
-        : items;
+      : results
```
[the remark this answers is not in the document]
Comment (Haroenv) [reply packages/react-instantsearch-core/src/connectors/connectMenu.js]:
```js
    let items;
    if (isFromSearch) {
      items = searchForFacetValuesResults[attribute].map(v => ({
        label: v.value,
        value: getValue(v.value, props, searchState, {
          ais: props.contextValue,
          multiIndexContext: props.indexContextValue,
        }),
        _highlightResult: { label: { value: v.highlighted } },
        count: v.count,
        isRefined: v.isRefined,
      }));
    } else {
      items = results
        .getFacetValues(attribute, {
          sortBy: searchable ? undefined : defaultSortBy,
        })
        .map(v => ({
          label: v.name,
          value: getValue(v.name, props, searchState, {
            ais: props.contextValue,
            multiIndexContext: props.indexContextValue,
          }),
          count: v.count,
          isRefined: v.isRefined,
        }));
    }
```

would this be your suggestion then?

[the remark this answers is not in the document]
Comment (samouss) [reply packages/react-instantsearch-core/src/connectors/connectMenu.js]: Yes, ternaries on multiple lines are hard to follow.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js [hunk 5]
```diff
@@ -416,7 +416,53 @@ describe('connectMenu', () => {
...
     });
 
-    it('when search for facets values is activated order the item by isRefined first', () => {
+    it('if not searchable: uses a static sortBy order', () => {
+      const results = {
+        getFacetValues: jest.fn(() => []),
+        getFacetByName: () => true,
+        hits: [],
+      };
+      results.getFacetValues.mockClear();
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js]: This call is useless since the implementation is defined above.

File: packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js [hunk 6]
```diff
@@ -416,7 +416,53 @@ describe('connectMenu', () => {
...
+      results.getFacetValues.mockImplementation(() => [
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js]: Why not inlining the mock implementation inside the declaration?

Comment (Haroenv) [reply packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js]: I didn't write this, only edited it. I tried out using a real helper, but it was a lot of boilerplate and much harder to set up by the way

File: packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js [hunk 7]
```diff
@@ -416,32 +416,80 @@ describe('connectMenu', () => {
...
     });
 
-    it('when search for facets values is activated order the item by isRefined first', () => {
+    it('if not searchable: uses a static sortBy order', () => {
       const results = {
         getFacetValues: jest.fn(() => []),
         getFacetByName: () => true,
         hits: [],
       };
       results.getFacetValues.mockClear();
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/__tests__/connectMenu.js]:
```suggestion

```

File: packages/react-instantsearch-core/src/core/indexUtils.js (hunk 3 above)
Comment (samouss) [reply packages/react-instantsearch-core/src/core/indexUtils.js]: @Haroenv Take one or the other suggestion is clearer that what we have right now.

File: packages/react-instantsearch-core/src/connectors/connectMenu.js [hunk 8]
```diff
@@ -135,18 +134,24 @@ export default createConnector({
...
+          multiIndexContext: props.indexContextValue,
+        }),
+        _highlightResult: { label: { value: v.highlighted } },
+        count: v.count,
+        isRefined: v.isRefined,
+      }));
+    } else {
+      items = results
+        .getFacetValues(attribute, {
+          sortBy: searchable ? undefined : defaultSortBy,
```
Comment (samouss) [inline packages/react-instantsearch-core/src/connectors/connectMenu.js]: Rather than relying on the default one from the Helper (which might change). We could specify the sort that we want to apply in this case.

Comment (Haroenv) [reply packages/react-instantsearch-core/src/connectors/connectMenu.js]: we will keep this as undefined, because the other widgets also pass undefined
